ingress/controllers/nginx/nginx/template.go

```diff
@@ -64,6 +64,27 @@ func (ngx *Manager) loadTemplate() error {
 }
 
 func (ngx *Manager) writeCfg(cfg config.Configuration, ingressCfg IngressConfig) (bool, error) {
+	var longestName int
+	var serverNames int
+	for _, srv := range ingressCfg.Servers {
```

Comment:
what's the tradeoff in being so precise vs just defaulting it to something high, documenting the limit, and not writing this code?